spec/support/event_helpers.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module EventHelpers
+  # pass in a block that returns true if the event list has the desired event(s)
+  def poll_for_matching_events!(prefixed_druid)
+    Dor::Services::Client.configure(url: Settings.dor_services.url, token: Settings.dor_services.token)
+    object_client = Dor::Services::Client.object(prefixed_druid)
+
+    Timeout.timeout(Settings.timeouts.workflow) do
```

Comment:
This is where the preservation timeout could be used; 

also, this could possibly become "poll_for_matching_replication_event(druid, version, s3key)"

Reply:
specific timeout values added